javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -690,7 +684,9 @@ public T get(int index) {
         if ((index < 0) || (index >= length())) {
             throw new IndexOutOfBoundsException("get(" + index + ")");
         } else {
-            return trie.get(index + indexShift).get();
+            final T[] leaf = trie.getLeaf(index);
+            final int leafIndex = lastDigit(trie.offset() + index);
+            return leaf[leafIndex];
```

Comment:
I could move the iterator at this stage, but when I'll introduce the `leading` and `trailing`, I would need to put it back.
I've had a separate `get` method at first also, but later decided that it's the same.
Will put both in `BMT` and move it back when the `PR` needs it, if that's ok.